src/NuGet.Clients/NuGet.CommandLine/Commands/PackCommand.cs

```diff
@@ -101,6 +107,16 @@ public override void ExecuteCommand()
             packArgs.BasePath = BasePath;
             packArgs.MsBuildDirectory = new Lazy<string>(() => MsBuildUtility.GetMsBuildDirectoryFromMsBuildPath(MSBuildPath, MSBuildVersion, Console).Value.Path);
 
+            if (!String.IsNullOrEmpty(PackagesDirectory))
+            {
+                packArgs.PackagesDirectory = Path.GetFullPath(PackagesDirectory);
+            }
+
+            if (!String.IsNullOrEmpty(SolutionDirectory))
+            {
+                packArgs.SolutionDirectory = String.IsNullOrEmpty(SolutionDirectory) ? null : Path.GetFullPath(SolutionDirectory);
```

Comment:
`String.IsNullOrEmpty(SolutionDirectory) ? null` you're already checking for this.